### ISSUE TITLE
Field extension bug

### ISSUE THREAD

**Issue by markuspf** [gap-system/gap#869]
### Observed behaviour

When creating a field extension over a field that is not a prime field:
```
    gap> f:= GF(4);;
    gap> x:= Indeterminate( f );;
    gap> g:= x^3 + x + 1;;
    gap> ext:= AlgebraicExtension( f, g );;
    gap> Length( Elements( ext ) );
    4096
    gap> Length( Set( Elements( ext ) ) );
    64
```

note that 

```
gap> ext;
<algebra-with-one of dimension 6 over GF(2^2)>
gap> LeftActingDomain(ext);
GF(2^2)
```
but the extension defined above only has dimension 3 over `GF(4)`. The Elements-Enumeration procedure assumes it gets a `GF(4)`-basis for `ext`, but it is actually getting a `GF(2)` basis.

### Expected behaviour

Field extensions over field extensions should work more or less in general.

**Comment by markuspf** [comment]
The field extension code itself was written by @hulpke as far as I can tell. Maybe we can quickly talk about this when he is in St Andrews.

**Comment by alex-konovalov** [comment]
The fix is now submitted by @hulpke as a part of #877

**Comment by alex-konovalov** [comment]
The fix for this and also for one more bug was merged in #877
